lib/utils/matrixUtils.js

```diff
@@ -0,0 +1,136 @@
+// Utility: norm and dot product
+// Useful util functions for common matrix tasks
+
+export function norm(vec) {
+  return Math.sqrt(vec.reduce((sum, x) => sum + x * x, 0));
+}
+
+export function dot(v1, v2) {
+  return v1.reduce((sum, x, i) => sum + x * v2[i], 0);
+}
+
+export function transpose(matrix) {
+  return matrix[0].map((_, i) => matrix.map((row) => row[i]));
+}
+
+export function scalarMultiply(matrix, scalar) {
+  return matrix.map((row) => row.map((val) => val * scalar));
+}
+
+export function identityMatrix(n) {
+  return Array.from({ length: n }, (_, i) =>
+    Array.from({ length: n }, (_, j) => (i === j ? 1 : 0))
+  );
+}
+
+export function multiplyMatrices(m1, m2) {
+  var result = [];
+  for (var i = 0; i < m1.length; i++) {
+    result[i] = [];
+    for (var j = 0; j < m2[0].length; j++) {
+      var sum = 0;
+      for (var k = 0; k < m1[0].length; k++) {
+        sum += m1[i][k] * m2[k][j];
+      }
+      result[i][j] = sum;
+    }
+  }
+  return result;
+}
+
+export function getMatrixParams(matrix) {
```

Comment:
This could be made more specific to be clear it's to get the cell representation in lengths + angles. Could the name and/or comments/doc be improved?